megamek/src/megamek/client/ratgenerator/MissionRole.java

```diff
@@ -418,6 +419,132 @@ public static Double adjustAvailabilityByRole(double avRating, Collection<Missio
 						avRating += avAdj[2] * (mRec.getSpeed() - (7 - mRec.getWeightClass()));					
 					}
 					break;
+				//Large craft roles
+				case CIVILIAN:
+				    if (!mRec.getRoles().contains(CIVILIAN)) {
+				        return null;
+				    }
+				    break;
+				case CARGO:
+				    if (mRec.getRoles().contains(CARGO)) {
+				        avRating += avAdj[2];
+                    } else if (mRec.getRoles().contains(CIVILIAN)) {
+                        avRating += avAdj[1];
+                    } else {
+                        avRating -= avAdj[2];
+                    }
+                    break;
+				case SUPPORT:
+                    if (mRec.getRoles().contains(SUPPORT)) {
+                        avRating += avAdj[2];
+                    } else if (mRec.getRoles().contains(TUG)) {
+                        avRating += avAdj[1];
+                    } else if (mRec.getRoles().contains(CARGO)
+                                || mRec.getRoles().contains(CIVILIAN)) {
+                        avRating += avAdj[0];
+                    } else {
+                        avRating -= avAdj[2];
+                    }
+                    break;
+				case TUG:
+                    if (!mRec.getRoles().contains(TUG)) {
+                        return null;
+                    }
+                    break;
+				case POCKET_WARSHIP:
+				    if (mRec.getRoles().contains(POCKET_WARSHIP)) {
+                        avRating += avAdj[2];
+                    } else if (mRec.getRoles().contains(ASSAULT)) {
+                        avRating += avAdj[0];
+                    } else {
+                        return null;
+                    }
+				    break;
+				case ASSAULT:
+                    if (mRec.getRoles().contains(ASSAULT)) {
+                        avRating += avAdj[2];
+                    } else if (mRec.getRoles().contains(POCKET_WARSHIP)) {
+                        avRating += avAdj[0];
+                    } else {
+                        return null;
+                    }
+                    break;
+				case MECH_CARRIER:
+				    if (mRec.getRoles().contains(MECH_CARRIER)) {
+                        avRating += avAdj[2];
+                    } else if (mRec.getRoles().contains(TROOP_CARRIER)) {
+                        avRating += avAdj[0];
+                    } else {
+                        return null;
+                    }
+                    break;
+				case ASF_CARRIER:
+				    if (mRec.getRoles().contains(ASF_CARRIER)) {
+                        avRating += avAdj[2];
+                    } else if (mRec.getRoles().contains(ASSAULT)) {
```

Comment:
Similarly, this should probably be implemented as a filter that returns null for any unit that does not contain ASF_CARRIER.

Reply:
Same as above, this is intentional since there are so few defined ASF carriers but so many assault ships carry about 6 fighters.